src/Functions/GatherUtils/Algorithms.h

```diff
@@ -404,6 +405,8 @@ static void sliceDynamicOffsetBoundedImpl(Source && src, Sink && sink, const ICo
             }
 
             writeSlice(slice, sink);
+        }else if(offset == 0){
```

Comment:
I don't expect we need this. When the index is non-constant, the input data is not controlled by the caller, and throwing an exception will be less convenient. See also `arrayElement`

Reply:
or can we not throw an exception when const offset is zero, to keep the  behavior consistent